### PR TITLE
Migrated com.hz.core.IExecutorService to com.hz.executor

### PR DESCRIPTION
**Pull request by pveentjer**

### CODE REVIEW

**Comment by mdogan** [comment]
What about `ExecutionCallback` and `MultiExecutionCallback` ?

**Comment by pveentjer** [comment]
Good point. 

ExecutionCallback is used by the ICompletableFuture. 

So it isn't specific to com.hz.executor.

WDYT? Better to have dependency on com.hz.core or on com.hz.executor?

**Comment by mdogan** [comment]
Hmm.. Will we keep `ICompletableFuture ` in 4.0 ? I don't know the outcome of https://hazelcast.atlassian.net/wiki/spaces/PM/pages/132026706/CompletableFuture yet.

File: hazelcast-spring/src/main/resources/hazelcast-spring-3.12.xsd [hunk 1]
```diff
@@ -2146,7 +2146,7 @@
             </xs:documentation>
             <xs:appinfo>
                 <tool:annotation>
-                    <tool:exports type="com.hazelcast.core.IExecutorService"/>
+                    <tool:exports type="com.hazelcast.executor.IExecutorService"/>
```
Comment (mdogan) [inline hazelcast-spring/src/main/resources/hazelcast-spring-3.12.xsd]: minor: no need to change 3.X XSD files.

**Comment by mmedenjak** [comment]
@mdogan removing `ICompletableFuture` is currently being worked on by @vbekiaris . I got the notion that it is going to happen and that there are no blocking issues at this point. Vassilis can provide more info.

**Comment by mdogan** [comment]
Then we can move execution callbacks to c.h.executor package too.

**Comment by vbekiaris** [comment]
`ICompletableFuture` is to be removed from 4.0 so `ExecutionCallback` can be moved to `com.hazelcast.executor` package

**Pull request closed without merging**